OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -294,9 +314,13 @@ public virtual IEnumerable<ActorInfo> BuildableItems()
 			if (!Enabled)
 				return Enumerable.Empty<ActorInfo>();
 			if (developerMode.AllTech)
-				return Producible.Keys;
-
-			return buildableProducibles;
+				return Producible.Keys.Where(a => GetProductionCost(a) < playerResources.GetCashAndResources() || IsInQueue(a));
```

Comment:
this is wrong as not it filters even when (!Info.PayUpFront || !Info.BlockedIfInsolvent) isn't false